src/main/java/de/tum/in/www1/artemis/domain/lecture/LectureUnitCompletion.java

```diff
@@ -0,0 +1,66 @@
+package de.tum.in.www1.artemis.domain.lecture;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+
+import javax.persistence.*;
+
+import de.tum.in.www1.artemis.domain.User;
+
+@Entity
+@Table(name = "lecture_unit_user")
+public class LectureUnitCompletion {
+
+    @EmbeddedId
+    @SuppressWarnings("unused")
+    private LectureUnitUserId id = new LectureUnitUserId();
+
+    @ManyToOne
+    @MapsId("userId")
+    private User user;
+
+    @ManyToOne
+    @MapsId("lectureUnitId")
+    private LectureUnit lectureUnit;
+
+    @Column(name = "completed_date")
+    private ZonedDateTime completedAt;
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    public LectureUnit getLectureUnit() {
+        return lectureUnit;
+    }
+
+    public void setLectureUnit(LectureUnit lectureUnit) {
+        this.lectureUnit = lectureUnit;
+    }
+
+    public ZonedDateTime getCompletedAt() {
+        return completedAt;
+    }
+
+    public void setCompletedAt(ZonedDateTime completedAt) {
+        this.completedAt = completedAt;
+    }
+
+    @Embeddable
+    @SuppressWarnings("unused")
```

Comment:
If this is unused why we can't remove it?

Reply:
The class itself is not unused, just the `userId` and `lectureUnitId`.